File: P14/APIPractice/app/src/main/java/com/example/apipractice/ApiClient.java
package com.example.apipractice;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
public class ApiClient {
    //    private static final String BASE_URL = "http://192.168.5.178:8080/";
    //emulator
    private static final String BASE_URL = "http://10.0.2.2:8080/";
    private static Retrofit retrofit;
    public static Retrofit getRetrofitInstance() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: Parkir/mobile/app/src/main/java/com/example/parkir/network/ApiClient.kt
package com.example.parkir.network
import com.google.gson.GsonBuilder
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.converter.scalars.ScalarsConverterFactory

// ApiClient.kt
object ApiClient {
    private const val BASE_URL = "http://192.168.5.178:8080/";

    // localhost untuk Android emulator
    // private const val BASE_URL = "http://10.0.2.2:8080/"

    private val gson = GsonBuilder()
        .setLenient()
        .create()

    private val retrofit = Retrofit.Builder()
        .baseUrl(BASE_URL)
        .addConverterFactory(ScalarsConverterFactory.create())
        .addConverterFactory(GsonConverterFactory.create(gson))
        .build()

    val apiService: ApiService = retrofit.create(ApiService::class.java)
}